packages/flutter/lib/src/widgets/form.dart

```diff
@@ -250,8 +250,16 @@ class FormState extends State<Form> {
 
   void _register(FormFieldState<dynamic> field) {
     _fields.add(field);
-    if (widget.autovalidateMode == AutovalidateMode.onUnfocus) {
-      field._focusNode.addListener(() => _updateField(field));
+    switch (widget.autovalidateMode) {
+      case AutovalidateMode.always:
+        if (field.widget.enabled) {
+          field.validate();
+        }
```

Comment:
One possible idea… maybe we _could_ do a post-frame callback here. Unlike [the previous fix](https://github.com/flutter/flutter/pull/156708#discussion_r1801554680), this wouldn't need a boolean flag to track the callback's status.

It wouldn't be quite as nice as the current setup, but it also might be less breaking.

Reply:
I tried to address the issues, but some tests are still failing. I'm considering reverting to this implementation with `didChangeDependencies` and documenting the technical debt needed to resolve the problems with the` Form`. I will attempt to write documentation or outline the technical debt for this.